test/suite/test_assert06.py

```diff
@@ -0,0 +1,403 @@
+        # again modify without a timestamp.
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_nots'] = 'value_nots'
+        self.session.commit_transaction()
+        c.close()
+
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_nots'] = 'value2'
+        self.apply_timestamps(2)
+        self.session.commit_transaction()
+        c.close()
+
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_nots'] = 'value_nots2'
+        self.session.commit_transaction()
+        c.close()
+
+        # We must move the oldest timestamp forward in order to alter.
+        # Otherwise alter closing the file will fail with EBUSY.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(2))
+
+        # Now alter the setting and make sure we detect incorrect usage.
+        self.session.alter(uri, cfg_on)
+
+        # Detect decreasing timestamp.
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_ts1'] = 'value5'
+        self.apply_timestamps(5)
+        self.session.commit_transaction()
+        c.close()
+
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_ts1'] = 'value4'
+        self.session.prepare_transaction(
+            'prepare_timestamp=' + timestamp_str(4))
+        self.session.timestamp_transaction(
+            'commit_timestamp=' + timestamp_str(4))
+        self.session.timestamp_transaction(
+            'durable_timestamp=' + timestamp_str(4))
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.commit_transaction(), msg_ooo)
+        c.close()
+
+        # Detect not using a timestamp.
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_ts1'] = 'value_nots3'
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.commit_transaction(), msg_usage)
+        c.close()
+
+        # Detect using a timestamp on the non-timestamp key.
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_nots'] = 'value_nots3'
+        self.session.commit_transaction()
+        c.close()
+
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_nots'] = 'value3'
+        self.apply_timestamps(3)
+        self.assertRaisesWithMessage(wiredtiger.WiredTigerError,
+            lambda: self.session.commit_transaction(), msg_usage)
+        c.close()
+        self.session.checkpoint()
+        
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+
+        self.assertEquals(c['key_ts1'], 'value5')
+        
+        self.assertEquals(c['key_nots'], 'value_nots3')
+        self.session.commit_transaction()
+
+        c.close()
+
```

Comment:
From here down it looks like the test is not quite finished. Key consistency should be able to be turned off, AFAIK.